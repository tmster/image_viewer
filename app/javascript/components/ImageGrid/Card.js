import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Card(props) {
  const { imageUrl, title, id } = props;
  return (
    <div className="col-lg-3 col-md-3 col-sm-6 col-xs-12">
      <Link to={`/image/${id}`}>
        <div className="card image-grid__card">
          <img
            className="card-img-top img-thumbnail image-grid__img"
            src={imageUrl}
            alt={title}
          />
        </div>
      </Link>
    </div>
  );
}

Card.propTypes = {
  imageUrl: PropTypes.string,
  id: PropTypes.string,
  title: PropTypes.string,
};

export default Card;
