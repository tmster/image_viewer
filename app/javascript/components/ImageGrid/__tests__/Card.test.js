import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter, Link } from 'react-router-dom';

import Card from '../Card';

it('renders all props', () => {
  const node = shallow(
    <Card id="0" title="Test" imageUrl="http://localhost" />
  );
  expect(node).toMatchSnapshot();
});

it('includes link to Mission scene', () => {
  const wrapper = mount(
    <MemoryRouter>
      <Card id="123" />
    </MemoryRouter>
  );
  expect(wrapper.find(Link).props().to).toBe('/image/123');
});
