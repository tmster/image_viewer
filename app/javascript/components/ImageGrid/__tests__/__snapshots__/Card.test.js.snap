// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all props 1`] = `
<div
  className="col-lg-3 col-md-3 col-sm-6 col-xs-12"
>
  <Link
    to="/image/0"
  >
    <div
      className="card image-grid__card"
    >
      <img
        alt="Test"
        className="card-img-top img-thumbnail image-grid__img"
        src="http://localhost"
      />
    </div>
  </Link>
</div>
`;
